cli/lib/service-generator.ts

```diff
@@ -0,0 +1,344 @@
+      let resData = formatJson(_resData);
+
+      if (_resData.startsWith('"Array<')) {
+        resData = formatJson(_resData.slice(7, _resData.length - 2));
+      }
+
+      apiList.push({
+        apiName,
+        method,
+        parameters: formatJson(parameters),
+        resData,
+        resType,
+      });
+      regRes = REG_API.exec(content);
+    }
+
+    forEach(apiList, ({ apiName, parameters, resData, resType }) => {
+      let pType = '{}';
+      let bType = 'RAW_RESPONSE';
+      if (resData !== '{}') {
+        if (resType === 'pagingList') {
+          bType = `IPagingResp<T_${apiName}_item>`;
+        } else if (resType === 'array') {
+          bType = `T_${apiName}_item[]`;
+        } else {
+          bType = `T_${apiName}_data`;
+        }
+
+        typeContent += `
+interface T_${apiName}_${['array', 'pagingList'].includes(resType) ? 'item' : 'data'} ${resData}\n`;
+      }
+      if (parameters !== '{}') {
+        typeContent += `
+interface T_${apiName}_params ${parameters}\n`;
+        pType = `T_${apiName}_params`;
+      }
+
+      appendContent += `
+export const ${apiName} = apiCreator<(p: ${pType}) => ${bType}>(apis.${apiName});
+        `;
+    });
+
+    fs.appendFileSync(typeFilePath, typeContent, 'utf8');
+    fs.appendFileSync(filePath, appendContent, 'utf8');
+    typeContent = '';
+    appendContent = '';
+  }
+
+  if (!isEnd) {
+    return;
+  }
+
+  if (isEnd) {
+    logSuccess('service data is updated, bye 👋');
+    resolve();
+  }
+};
+
+const getSwaggerData = () => {
+  return http
+    .get(`https://api.github.com/repos/${username}/${repository}/contents/${swaggerFilePath}`)
+    .set('content-type', 'application-json')
+    .set('User-Agent', '')
+    .then((response) => {
+      logInfo('get swagger data successfully!');
+      const content = decode(response?.body?.content);
+      return content ? JSON.parse(content) : '';
+    })
+    .catch((err) => {
+      logError('fail to get swagger data: ', err);
+      return false;
+    });
+};
+
+const extractHandle = async () => {
+  await new Promise<void>((resolve) => {
+    walker({
+      root: workDir,
+      dealFile: (...args) => {
+        autoGenerateService.apply(null, [...args, resolve]);
+      },
+    });
+  });
+};
+
+const generate = async () => {
+  swaggerData = await getSwaggerData();
+  if (keys(swaggerData?.paths)?.length) {
+    extractHandle();
+  } else {
+    logError('It is an empty swagger!');
+    process.exit(1);
+  }
+};
+
+export default ({
+  workDir: _workDir,
+  repository: _repository,
+  username: _username,
+  filePath: _filePath,
+}: {
+  workDir: string;
+  repository: string;
+  username: string;
+  filePath: string;
+}) => {
+  try {
+    workDir = _workDir || process.cwd();
+    repository = _repository || 'erda';
+    username = _username || 'erda-project';
+    swaggerFilePath = _filePath || 'pkg/swagger/oas3/testdata/swagger_all.json';
+
+    generate();
```

Comment:
You can move extractHandle and generate logic into here, so you could remove these global variable.